test/tools/cluster.sh

```diff
@@ -0,0 +1,198 @@
+    elif [[ "$COMMAND" == "stop" ]]; then
+        TOPOLOGY=$2
+        if is_replica $TOPOLOGY; then
+            log 'stopping replica'
+            mlaunch stop --dir "./data/$PORT_REPLICA"
+            if [ $? -ne 0 ]; then
+                echo "error: cannot stop replica, not initalized"
+            fi
+        elif is_sharded $TOPOLOGY; then
+            log 'stopping sharded'
+            mlaunch stop --dir "./data/$PORT_SHARDED"
+            if [ $? -ne 0 ]; then
+                echo "error: cannot stop sharded, not initalized"
+            fi
+        elif is_single $TOPOLOGY; then
+            log 'stopping single'
+            mlaunch stop --dir "./data/$PORT_SINGLE"
+            if [ $? -ne 0 ]; then
+                echo "error: cannot stop single, not initalized"
+            fi
+        else
+            cluster stop replica
+            cluster stop sharded
+            cluster stop single
+        fi
+    elif [[ "$COMMAND" == "prune" ]]; then
+        TOPOLOGY=$2
+        if is_replica $TOPOLOGY; then
+            log 'pruning replica'
+            cluster stop replica
+            log 'removing replica'
+            rm -rf "./data/$PORT_REPLICA";
+        elif is_sharded $TOPOLOGY; then
+            log 'pruning sharded'
+            cluster stop sharded
+            log 'removing sharded'
+            rm -rf "./data/$PORT_SHARDED";
+        elif is_single $TOPOLOGY; then
+            log 'pruning server'
+            cluster stop single
+            log 'removing server'
+            rm -rf "./data/$PORT_SINGLE";
+        else
+            cluster prune replica
+            cluster prune sharded
+            cluster prune single
+        fi
+      elif [[ "$COMMAND" == "kill" ]]; then
+        TOPOLOGY=$2
+        if is_replica $TOPOLOGY; then
+            log 'killing replica'
+            PORT1=$PORT_REPLICA
+            PORT2=$(( $PORT1 + 1 ))
+            PORT3=$(( $PORT2 + 1 ))
+            PORT4=$(( $PORT3 + 1 ))
+            npx -q fkill-cli :"$PORT1" :"$PORT2" :"$PORT3" :"$PORT4" -fs
+        elif is_sharded $TOPOLOGY; then
+            log 'killing sharded'
+            PORT1=$PORT_SHARDED
+            PORT2=$(( $PORT1 + 1 ))
+            PORT3=$(( $PORT2 + 1 ))
+            PORT4=$(( $PORT3 + 1 ))
+            PORT5=$(( $PORT4 + 1 ))
+            PORT6=$(( $PORT5 + 1 ))
+            PORT7=$(( $PORT6 + 1 ))
+            npx -q fkill-cli :"$PORT1" :"$PORT2" :"$PORT3" :"$PORT4" :"$PORT5" :"$PORT6" :"$PORT7" -fs
+        elif is_single $TOPOLOGY; then
+            log 'killing single'
+            npx -q fkill-cli :$PORT_SINGLE -fs
+        else
+            cluster kill replica
+            cluster kill sharded
+            cluster kill single
+        fi
+      elif [[ "$COMMAND" == "test" ]]; then
+        TOPOLOGY=$2
+        if is_replica $TOPOLOGY; then
+            cluster start replica && cluster uri replica && npm run test-nolint --if-present -- "${@:3}" && npm run check:test --if-present -- "${@:3}"
```

Comment:
Added "${@:3}" so that you can run:

`cluster test single --bail` ect and extra args pass through to mocha